sdks/python/apache_beam/yaml/yaml_provider.py

```diff
@@ -700,9 +700,25 @@ def __init__(self, transforms, mappings, underlying_provider, defaults=None):
     for transform in transforms.keys():
       if transform not in mappings:
         raise ValueError(f'Missing transform {transform} in mappings.')
-    self._mappings = mappings
+    self._mappings = self.get_mappings_from_transform(mappings)
```

Comment:
Maybe call this "validate_mappings" or "expand_mappings" to make the intent more clear?